src/common/mfu_io.c

```diff
@@ -628,6 +628,54 @@ int mfu_remove(const char* path)
  * Links
  ****************************/
 
+/* emulates readlink with dfs_lookup, dfs_get_symlink_value */
+ssize_t daos_readlink(const char* path, char* buf, size_t bufsize, mfu_file_t* mfu_file)
+{
+#ifdef DAOS_SUPPORT
+    char* name     = NULL;
+    char* dir_name = NULL;
+    parse_filename(path, &name, &dir_name);
+    assert(dir_name);
+
+    int rc = 0;
+    daos_size_t got_size = (daos_size_t) bufsize;
+
+    /* Lookup the parent directory first, since it is likely cached */
+    dfs_obj_t* parent = daos_hash_lookup(dir_name, mfu_file);
+    if (parent == NULL) {
+        errno = ENOENT;
+        rc = -1;
```

Comment:
Looks like the code below is return ``got_size``.  Do you mean to set that to -1 instead?

Reply:
Yes, thank you. Gotta love copy-paste!